File: backend/app/schemas/weather.py
from datetime import datetime
from pydantic import BaseModel


class WeatherPoint(BaseModel):
    time: datetime
    temp_c: float
    feels_like_c: float | None = None
    wind_ms: float
    wind_gust_ms: float | None = None
    precip_mm: float
    precip_probability: float | None = None
    humidity_pct: float
    pressure_hpa: float | None = None
    cloudiness_pct: int | None = None
    description: str
    has_precipitation: bool = False


class WeatherForecast(BaseModel):
    site_id: str
    lat: float
    lon: float
    fetched_at: datetime
    points: list[WeatherPoint]
    source: str

File: frontend/src/types/weather.ts
export interface WeatherPoint {
  time: string;
  temp_c: number;
  feels_like_c: number | null;
  wind_ms: number;
  wind_gust_ms: number | null;
  precip_mm: number;
  precip_probability: number | null;
  humidity_pct: number;
  pressure_hpa: number | null;
  cloudiness_pct: number | null;
  description: string;
  has_precipitation: boolean;
}

export interface WeatherForecast {
  site_id: string;
  lat: number;
  lon: number;
  fetched_at: string;
  points: WeatherPoint[];
  source: string;
}
